### PR TITLE
Join case new enhancement

### PR DESCRIPTION
**Pull request by nitish-beehyv**
Feature tickents: #1993 and #2387

## Summary by CodeRabbit

- **New Features**
	- Enhanced modal management for response submissions in the Tasks component.
	- Added functionality for document uploads during response submissions.
	- Improved handling of user roles and state management in the Join Case component.

- **Bug Fixes**
	- Refined error handling for validation codes to provide clearer user feedback.

- **Refactor**
	- Simplified state management in the Join Case component by removing unnecessary state variables.
	- Streamlined logic for enabling/disabling form submission based on user selections.
	- Updated styling and layout of the Case Lock modal for improved user experience.

### CODE REVIEW

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/cases/src/pages/employee/JoinCaseHome.js [hunk 1]
```diff
@@ -1090,126 +876,48 @@ const JoinCaseHome = ({ refreshInbox, setShowSubmitResponseModal, setResponsePen
...
-                    </div>
+                    <CardLabel className="case-input-label">{`${t(JoinHomeLocalisation.PARTY_PARTIES)}`}</CardLabel>
+                    <Dropdown
+                      t={t}
+                      option={parties}
+                      selected={party}
+                      optionKey={"fullName"}
+                      select={(e) => setParty(e)}
+                      freeze={true}
+                      disable={true}
```
[the remark this answers is not in the document]
Comment (nitish-beehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/cases/src/pages/employee/JoinCaseHome.js:916]: checked

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/cases/src/pages/employee/JoinCaseHome.js [hunk 2]
```diff
@@ -1465,17 +1173,17 @@ const JoinCaseHome = ({ refreshInbox, setShowSubmitResponseModal, setResponsePen
                   className={"selector-button-primary"}
                   label={
                     caseDetails?.status === "PENDING_RESPONSE" && selectedParty?.isRespondent
-                      ? "Submit Response"
-                      : !isAttendingHearing && nextHearing
-                      ? "Confirm attendance in summon"
+                      ? t("SUBMIT_RESPONSE_TEXT")
+                      : !isAttendingHearing && nextHearing && selectedParty?.isRespondent
+                      ? t("CONFIRN_SUMMON_NOTICE_RECEIPT")
```
[the remark this answers is not in the document]
Comment (nitish-beehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/cases/src/pages/employee/JoinCaseHome.js:1178]: checked

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/cases/src/pages/employee/JoinCaseHome.js [hunk 3]
```diff
@@ -1090,126 +876,48 @@ const JoinCaseHome = ({ refreshInbox, setShowSubmitResponseModal, setResponsePen
...
+                      // t={t}
+                      style={{ width: "100%" }}
+                      type={"text"}
+                      name="barRegNumber"
+                      value={barRegNumber}
+                      onChange={(e) => {
+                        setBarRegNumber(e.target.value);
+                        setBarDetails([]);
                       }}
+                      disable={userType?.value === "Litigant" ? false : true}
```
[the remark this answers is not in the document]
Comment (nitish-beehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/cases/src/pages/employee/JoinCaseHome.js:899]: checked

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/cases/src/pages/employee/JoinCaseHome.js [hunk 4]
```diff
@@ -1090,126 +876,48 @@ const JoinCaseHome = ({ refreshInbox, setShowSubmitResponseModal, setResponsePen
...
-                      formData={formData}
-                      config={documentUploaderConfig}
-                      onSelect={(e, p) => {
-                        setFormData({
-                          [documentUploaderConfig.key]: p,
-                        });
+
+              <React.Fragment>
+                <LabelFieldPair className="case-label-field-pair">
+                  <CardLabel className="case-input-label">{`${"BAR registration"}`}</CardLabel>
```
[the remark this answers is not in the document]
Comment (nitish-beehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/cases/src/pages/employee/JoinCaseHome.js:887]: fixed

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/cases/src/pages/employee/JoinCaseHome.js [hunk 5]
```diff
@@ -2708,7 +2179,7 @@ const JoinCaseHome = ({ refreshInbox, setShowSubmitResponseModal, setResponsePen
       updatedHearing.attendees.push({
         name: formatFullName(name) || "",
         individualId: individualId,
-        type: "Respondent",
+        type: type,
```
[the remark this answers is not in the document]
Comment (nitish-beehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/cases/src/pages/employee/JoinCaseHome.js:2182]: fixed

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/cases/src/pages/employee/JoinCaseHome.js [hunk 6]
```diff
@@ -1697,28 +1404,16 @@ const JoinCaseHome = ({ refreshInbox, setShowSubmitResponseModal, setResponsePen
...
-  };
-
   const onProceed = useCallback(async () => {
     if (step === 0) {
       if (caseDetails?.cnrNumber) {
         if (userType?.value === "Litigant") {
           const isFound = caseDetails?.litigants?.find((item) => item.individualId === individualId);
           if (isFound !== undefined) {
-            setStep(8);
+            setSelectedParty({ isRespondent: isFound?.partyType?.includes("respondent") ? true : false });
```
[the remark this answers is not in the document]
Comment (nitish-beehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/cases/src/pages/employee/JoinCaseHome.js:1413]: fixed

**Review by anirudh-0 — APPROVED** [review]

**Pull request merged**